examples/Example.elm

```diff
@@ -0,0 +1,149 @@
+module Example exposing (..)
+
+import Html
+import Json.Decode as Decode
+import Json.Encode as Encode
+import Kinto
+
+
+-- Define what our Kinto records will look like for a Todo list.
+
+
+type alias Todo =
+    { id : String
+    , title : Maybe String
+    , description : Maybe String
+    , last_modified : Int
+    }
+
+
+
+-- And how to decode from a Kinto record to a Todo...
+
+
+decodeTodo : Decode.Decoder Todo
+decodeTodo =
+    (Decode.map4 Todo
+        (Decode.field "id" Decode.string)
+        (Decode.maybe (Decode.field "title" Decode.string))
+        (Decode.maybe (Decode.field "description" Decode.string))
+        (Decode.field "last_modified" Decode.int)
+    )
+
+
+
+-- And how to encode Todo data to send to Kinto.
+
+
+encodeData : Maybe String -> Maybe String -> Encode.Value
+encodeData title description =
+    Encode.object
+        [ ( "title", Encode.string (Maybe.withDefault "" title) )
+        , ( "description", Encode.string (Maybe.withDefault "" description) )
+        ]
+
+
+
+-- Configure a Kinto client: the server url and authentication
+
+
+client : Kinto.Client
+client =
+    Kinto.client
+        "https://kinto.dev.mozaws.net/v1/"
+        (Kinto.Basic "test" "test")
+
+
+
+-- Set up the resource we're interested in: in this case, Kinto records, stored
+-- in the "test-items" collection, in the "default" bucket.
+-- We give it the `decodeTodo` decoder so the data we get back from the Kinto
+-- server will be properly decoded into our `Todo` record.
+
+
+recordResource : Kinto.Resource Todo
+recordResource =
+    Kinto.recordResource "default" "test-items" decodeTodo
+
+
+
+-- Add a new Todo
+
+
+addTodo : Maybe String -> Maybe String -> Cmd Msg
+addTodo title description =
```

Comment:
The [Yesod framework](http://www.yesodweb.com/) solves this problem by defining an `Entity a` type that is essentially `(Id, a)`. So here you would have a `Todo` record, but a `Todo` that was in the DB would be an `Entity Todo`. They write about it [here](http://www.yesodweb.com/book/persistent#persistent_insert).